### ISSUE TITLE
Consolidate Wyze API Infrastructure

### ISSUE THREAD

**Issue by carTloyal123** [shauntarves/wyze-sdk#178]
Hi Shaun,

Long time supporter of the project and have been using it for a project of my own. I have noticed that there are a few, at least three, major Wyze projects that have to make Wyze API calls and thus have built up Wyze API code to support those calls to various endpoints. 

I am wondering if you would be interested in breaking out the Wyze API modules of this project to support/use/contribute to a central Wyze API repo. I think the benefits are pretty clear that Wyze API updates will be centralized and any existing future projects can use the repo as a drop in for calling to Wyze services. My project is here: [Cryze](https://github.com/carTloyal123/cryze) and I am also referring to [Docker Wyze Bridge](https://github.com/mrlt8/docker-wyze-bridge) and your project as well.

Love to hear your thoughts! Thank you!

**Comment by KyGuy2002** [comment]
I think this would be a great idea.  The wyze apis are so complicated and scattered, it is impossible for one project to implement and maintain everything.  I am currently working on adding a ton of improvements to wyzeapy (used by the ha-wyzeapi homeassitant integration), and don't want to repeat work that's been done here.

**Comment by carTloyal123** [comment]
any thoughts @shauntarves ?

**Comment by shauntarves** [comment]
I agree, this would be awesome. To be clear, I built this library to serve as a "low-level" SDK to build solutions on top of. That sounds pretty similar to what you're proposing, so I'm curious what kind of architecture/system you are thinking about.
